src/features/unified-share-modal/__tests__/UnifiedShareForm.test.js

```diff
@@ -629,39 +628,63 @@ describe('features/unified-share-modal/UnifiedShareForm', () => {
-            ${false}                                | ${defaultContacts.map(({ email }) => email)} | ${false}    | ${'not fetch'}    | ${'collab restrictions change but justification is not allowed'}
-            ${true}                                 | ${['unmatched_email@example.com']}           | ${false}    | ${'not fetch'}    | ${'collab restrictions change but no restricted collabs are present'}
-            ${true}                                 | ${defaultContacts.map(({ email }) => email)} | ${true}     | ${'fetch'}        | ${'collab restrictions change, justification is allowed and restricted collabs are present'}
+            isCollabRestrictionJustificationAllowed | restrictedCollabEmails                           | restrictedGroups | conditionDescription
+            ${false}                                | ${[]}                                            | ${[]}            | ${'collab restrictions do not change'}
+            ${false}                                | ${defaultUserContacts.map(({ email }) => email)} | ${[]}            | ${'collab restrictions change but justification is not allowed'}
+            ${true}                                 | ${['unmatched_email@example.com']}               | ${[]}            | ${'collab restrictions change but no restricted collabs are present'}
         `(
-            'should $actionDescription justification reasons when $conditionDescription',
-            ({ isCollabRestrictionJustificationAllowed, restrictedCollabEmails, shouldFetch }) => {
+            'should not fetch justification reasons when $conditionDescription',
+            ({ isCollabRestrictionJustificationAllowed, restrictedCollabEmails, restrictedGroups }) => {
                 const getJustificationReasons = jest.fn().mockResolvedValue({});
                 const wrapper = getWrapper({
                     getJustificationReasons,
                     isCollabRestrictionJustificationAllowed: false,
                     item: defaultItem,
                     restrictedCollabEmails: [],
+                    restrictedGroups: [],
                 });
 
                 wrapper.instance().updateInviteCollabsContacts(defaultContacts);
                 wrapper.setProps({
                     isCollabRestrictionJustificationAllowed,
                     restrictedCollabEmails,
+                    restrictedGroups,
                 });
 
-                if (shouldFetch) {
-                    expect(getJustificationReasons).toHaveBeenCalledTimes(1);
-                    expect(getJustificationReasons).toHaveBeenCalledWith(
-                        defaultItem.typedID,
-                        JUSTIFICATION_CHECKPOINT_EXTERNAL_COLLAB,
-                    );
-                } else {
-                    expect(getJustificationReasons).toHaveBeenCalledTimes(0);
-                }
+                expect(getJustificationReasons).toHaveBeenCalledTimes(0);
```

Comment:
Split these in two to avoid adding more table columns and to remove the conditional logic